### ISSUE TITLE
Binder creates its own KTM instead of injecting Boot's

### ISSUE THREAD

**Issue by garyrussell** [spring-cloud/spring-cloud-stream-binder-kafka#623]
Also, if there is no existing KTM bean the one the binder creates should have 

    setTransactionSynchronization(AbstractPlatformTransactionManager.SYNCHRONIZATION_ON_ACTUAL_TRANSACTION)

So that the producer will participate in the consumer's transaction, for exactly once semantics.

Work around is to add `TransactionSynchronizationManager.setActualTransactionActive(true);` to the `@StreamListener`.

**Comment by garyrussell** [comment]
Problem is in SIK - but leaving this since the KTM should be configurable.

**Comment by garyrussell** [comment]
The reason for opening this has been fixed in S-I-K 3.1.2 https://github.com/spring-projects/spring-integration-kafka/pull/261

So, while this should still be done, it is not urgent.

**Comment by sobychacko** [comment]
From the binder, we can now use `ListenerContainerCustomizer` to configure a ` KafkaTransactionManager`. 
See this [answer](https://stackoverflow.com/a/57379903/2070861) from @garyrussell on this Stack Overflow issue. 
Essentially, you can use the `ListenerContainerCustomizer` in the following way to customize a Kafka transaction manager:
```
@Bean
public ListenerContainerCustomizer<AbstractMessageListenerContainer<byte[], byte[]>> customizer() {
    return (container, dest, group) -> {
        KafkaTransactionManager<?, ?> tm = (KafkaTransactionManager<?, ?>) container.getContainerProperties()
                .getTransactionManager();
        tm.setTransactionSynchronization(AbstractPlatformTransactionManager.SYNCHRONIZATION_ON_ACTUAL_TRANSACTION);
    };
}
```

Closing the issue.
